biopet-tools/src/main/scala/nl/lumc/sasc/biopet/tools/refflatstats/RefflatStats.scala

```diff
@@ -0,0 +1,224 @@
+package nl.lumc.sasc.biopet.tools.refflatstats
+
+import java.io.{File, PrintWriter}
+
+import htsjdk.samtools.reference.IndexedFastaSequenceFile
+import nl.lumc.sasc.biopet.utils.intervals.{BedRecord, BedRecordList}
+import nl.lumc.sasc.biopet.utils.{FastaUtils, ToolCommand}
+import picard.annotation.{Gene, GeneAnnotationReader}
+
+import scala.collection.JavaConversions._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future, TimeoutException}
+
+/**
+  * Created by pjvan_thof on 1-5-17.
+  */
+object RefflatStats extends ToolCommand {
+
+  case class Args(refflatFile: File = null,
+                  referenceFasta: File = null,
+                  geneOutput: File = null,
+                  transcriptOutput: File = null,
+                  exonOutput: File = null,
+                  intronOutput: File = null)
+      extends AbstractArgs
+
+  class OptParser extends AbstractOptParser {
+    opt[File]('a', "annotationRefflat") required () maxOccurs 1 valueName "<file>" action {
+      (x, c) =>
+        c.copy(refflatFile = x)
+    }
+    opt[File]('R', "referenceFasta") required () maxOccurs 1 valueName "<file>" action { (x, c) =>
+      c.copy(referenceFasta = x)
+    }
+    opt[File]('g', "geneOutput") required () maxOccurs 1 valueName "<file>" action { (x, c) =>
+      c.copy(geneOutput = x)
+    }
+    opt[File]('t', "transcriptOutput") required () maxOccurs 1 valueName "<file>" action {
+      (x, c) =>
+        c.copy(transcriptOutput = x)
+    }
+    opt[File]('e', "exonOutput") required () maxOccurs 1 valueName "<file>" action { (x, c) =>
+      c.copy(exonOutput = x)
+    }
+    opt[File]('i', "intronOutput") required () maxOccurs 1 valueName "<file>" action { (x, c) =>
+      c.copy(intronOutput = x)
+    }
+  }
+
+  /**
+    * Program will prefix reads with a given seq
+    *
+    * @param args the command line arguments
+    */
+  def main(args: Array[String]): Unit = {
+    logger.info("Start")
+
+    val argsParser = new OptParser
+    val cmdArgs: Args =
+      argsParser.parse(args, Args()) getOrElse (throw new IllegalArgumentException)
+
+    //Sets picard logging level
+    htsjdk.samtools.util.Log
+      .setGlobalLogLevel(htsjdk.samtools.util.Log.LogLevel.valueOf(logger.getLevel.toString))
+
+    logger.info("Reading refflat file")
+
+    val geneReader = GeneAnnotationReader.loadRefFlat(
+      cmdArgs.refflatFile,
+      FastaUtils.getCachedDict(cmdArgs.referenceFasta))
+
+    val futures = geneReader.getAll.map(generateGeneStats(_, cmdArgs.referenceFasta)).toList
```

Comment:
Referring to the object type, see also http://www.scala-lang.org/api/2.11.4/index.html#scala.concurrent.Future